Sources/HAP/Endpoints/characteristics().swift

```diff
@@ -25,18 +34,53 @@ func characteristics(device: Device) -> Application {
                     let characteristic = device.accessories.first(where: {$0.aid == path[0]})?.services.flatMap({$0.characteristics.filter({$0.iid == path[1]})}).first
                     else {
                         // @fixme: hc sets status to StatusServiceCommunicationFailure instead
-                        return .notFound
+                        serialized.append(["aid": path[0],"iid": path[1],"status": HAPStatusCodes.resourceDoesNotExist.rawValue])
+                        multiStatusResponse = true
+                        break
+                }
+                
+                var body = ["aid": path[0],"iid": path[1],"value": characteristic.getValue() ?? NSNull()]
+                if meta {
+                    if let maxValue = characteristic.maxValue {
+                        body["maxValue"] = maxValue.jsonValueType
+                    }
+                    if let minValue = characteristic.minValue {
+                        body["minValue"] = minValue.jsonValueType
+                    }
+                    if let unit = characteristic.unit {
+                        body["unit"] = unit.jsonValueType
+                    }
+                    if let minStep = characteristic.minStep {
+                        body["minStep"] = minStep.jsonValueType
+                    }
+                    if let maxLen = characteristic.maxLength {
+                        body["maxLen"] = maxLen.jsonValueType
+                    }
+                }
+                if perms {
+                    body["perms"] = characteristic.permissions.map { $0.rawValue }
+                }
+                if type {
+                    body["type"] = characteristic.type.rawValue
+                }
+                if ev {
+                    body["ev"] = characteristic.permissions.contains(.events)
+                }
+                
+                serialized.append(body)
+            }
+            
+            if multiStatusResponse {
+                for (index,element) in serialized.enumerated() {
+                    if element["status"] == nil {
+                        serialized[index]["status"] = HAPStatusCodes.success.rawValue
```

Comment:
I'm not familiar with section 5.7 of the HAP spec yet. Do we only need to set the `status` when we're doing a `multiStatus` response?

Reply:
The specs are not clear on this. The examples, hovewer, suggest that the status should be omitted where there are no errors.